### PR TITLE
Improve efficiency of test-one-license

### LINKED ISSUE #1760: test-one-license can be more efficient
The [test-one-license script](https://github.com/spdx/license-list-XML/blob/main/test-one-license) can be used to test a single license ID against its test text file.

Currently, it does this by renaming all other licenses to add a ".skip" extension, then runs the license list publisher's testing tool against the single remaining license. That can be slow, I believe because the "validate-canonical-match" process may do some additional work that isn't necessary when just testing one license.

However, the license list publisher tool now appears to allow [testing just a single license](https://github.com/spdx/licenseListPublisher#syntax) via the function called from the command line:

> The command line interface of the licenseListPublisher can be used like this:

```
java -jar licenseListPublisher.jar <function> <parameters> 
```

> Where the following functions and parameters are supported:

```
TestLicenseXML licenseXmlFile textFile
   licenseXmlFile XML - file to test
   textFile - Text file which should match the the license text for the licenseXmlFile
```

The test-one-license script could be significantly shortened to just be a call to this function. I'll put together a PR to handle this.

### PR DESCRIPTION
**Pull request by swinslow**
Fixes #1760

Signed-off-by: Steve Winslow <steve@swinslow.net>

### CODE REVIEW

**Comment by swinslow** [comment]
@goneall Absolutely no urgency on this one  :)  But I'd welcome your thoughts on this one at some point!

**Review by goneall — CHANGES_REQUESTED** [review]
This should work with the exception noted below on the test directory parameter - the LicenseListPublisher tools has been updated to test a single license.

The one additional suggestion - you may want to validate the download of the license list publisher using the GPG signature - see https://github.com/spdx/license-list-XML/blob/50f2139df761669d6244825699a08446a547d56b/Makefile#L58

File: test-one-license [hunk 1]
```diff
@@ -1,22 +1,9 @@
...
 
-# Run the test suite with just the XML file for the license
-# we want to test reamining.
-make | tee "test.log"
-
-# Remove the skip suffix from all XML files.
-while IFS= read -d '' -r file; do
-  mv "$file" "$(dirname "$file")/$(basename "$file" "$SKIP_SUFFIX")"
-done < <(find src -type f -name "*$SKIP_SUFFIX" -print0)
+java -jar ${PUBLISHER} TestLicenseXML "src/${@}.xml" "test/simpleTestForGenerator/${@}.txt"
```
[the remark this answers is not in the document]
Comment (swinslow) [reply test-one-license:9]: Hi @goneall, I tried changing this to remove the filename from the last parameter, and I'm getting an error message:

```
Test file simpleTestForGenerator is not a valid file
Usage:
TestLicenseXML licenseXmlFile textFile
   licenseXmlFile XML - file to test
   textFile - Text file which should match the the license text for the licenseXmlFile
```

Looking at the LicenseListPublisher code, if I'm reading it correctly, I think it checks to confirm that the specified file actually is a file: see https://github.com/spdx/LicenseListPublisher/blob/f2f481776d149c7d610a2777874abfda87d7dae3/src/org/spdx/licenselistpublisher/LicenseXmlTester.java#L78-L82

I might be misunderstanding, but it looks to me like that command at least is expecting the specific test text file to be provided as an argument?

**Comment by swinslow** [comment]
Thanks @goneall! For the file vs. directory argument, I just replied inline to the comment above.

For validating the publisher GPG signature, the current PR does the following:
* uses `awk` to extract the TOOL_VERSION from the most current Makefile
* calls `make licenseListPublisher-%.jar-valid` for the TOOL_VERSION as part of downloading the publisher, if it is not already present on disk -- see https://github.com/spdx/license-list-XML/blob/05639a5ddc9294d998863945b06d94aec9ca0c81/test-one-license#L7

Is the call to e.g. `make licenseListPublisher-2.2.7.jar-valid` sufficient to do the GPG validation? It looks like it's occurring when I run it locally, but not sure if I'm missing something.

File: test-one-license (hunk 1 above)
[the remark this answers is not in the document]
Comment (goneall) [reply test-one-license:9]: > I might be misunderstanding, but it looks to me like that command at least is expecting the specific test text file to be provided as an argument?

Ahh - I was looking at a different utility that will also test a single file.  You are correct, this command expects a single test file.

My bad - I was looking at the `LicenseRDFAGenerator` which can also take a single file as the input XML but takes a directory for the test files.

**Comment by goneall** [comment]
> Is the call to e.g. `make licenseListPublisher-2.2.7.jar-valid` sufficient to do the GPG validation? It looks like it's occurring when I run it locally, but not sure if I'm missing something.

You're correct on this one as well, the makefile does validate.  Another miss on the review - I guess I reviewed it a bit too quickly :(

**Review by goneall — APPROVED** [review]
LGTM

**Comment by swinslow** [comment]
No worries at all, thank you @goneall!

**Pull request merged**